virttest/qemu_vm.py

```diff
@@ -2651,7 +2651,9 @@ def add_secure_guest_object(params):
             add_qemu_option(devices, "k", [attr_info])
 
         # Add options for all virtio devices
-        virtio_devices = filter(lambda x: re.search(r"(?:^virtio-)|(?:^vhost-)",
+        virtio_dev_filter = params.get("virtio_dev_filter",
+                                       "(?:^virtio-)|(?:^vhost-)")
+        virtio_devices = filter(lambda x: re.search(virtio_dev_filter,
```

Comment:
In this way, user define a whitelist. However, i think it is more convenient to define a blacklist.  As I think user does not know all the devices (such as virtio-net that are defined in framework), but they know the devices that be used in the case but do not need the default virtio option.  What do you think about this ?

By the way,  it is better to change the the comment with L2653, not `all virtio devices` now :)

Reply:
Sure, I already updated L2653, and after I rethought it, the filter can also define a blacklist, you can check the following test results.